### PR TITLE
services/keystore: consolidate authentication section in the spec

### PR DESCRIPTION
**Pull request by howardtw**
This PR updates the keystore spec with more details about how the
authentication works. It also combines two sections together.

It piggybacks a rename of a file containing a helper function. It was an
accident that I named that file as `main_test.go`

### CODE REVIEW

File: services/keystore/spec.md [hunk 1]
```diff
@@ -18,16 +18,34 @@ don’t anticipate a lot of requests to the keystore from each user, we
...
 auth token to the client server.
 
-Clients will have to configure a API endpoint on their servers used for
-authentication when booting up the keystore. Please refer to [this section](#required-changes-in-client-server)
-for more details.
+<img src=attachments/2019-07-10-keystore-auth.png>
 
-An auth token needs to be passed with all requests, and that's why there
-is no need for a userid argument to the endpoints below.
+Keystore will forward every header field to the designated endpoint on the
```
Comment (tomquisel) [inline services/keystore/spec.md]: Maybe we should only forward `Authorization` and `Cookie` headers? Headers like `Host`, `Accept`, and caching headers will be wrong or confusing for the auth server I think.

File: services/keystore/spec.md [hunk 2]
```diff
@@ -18,16 +18,34 @@ don’t anticipate a lot of requests to the keystore from each user, we
...
+client server with an extra header field *X-Forwarded-For* specifying the
+request's origin. At this moment, keystore forwards incoming requests by using
+HTTP GET method. We plan on adding the support for clients who use GraphQL to
+authenticate in the future.
 
-<img src=attachments/2019-04-24-keystore-auth-flows.png>
+Clients are expected to put their auth tokens in one of the request header
+fields. For example, those who use a bearer token to autheticate should have an
```
Comment (tomquisel) [inline services/keystore/spec.md]: authenticate :)

File: services/keystore/spec.md (hunk 1 above)
Comment (howardtw) [reply services/keystore/spec.md]: I wasn't sure whether clients would only use `Authorization` header to do authentication tho.

Comment (tomquisel) [reply services/keystore/spec.md]: Yeah, agreed. I think `Authorization` and `Cookie` will cover most use cases. We can always add more later if another need becomes clear.

Comment (howardtw) [reply services/keystore/spec.md]: I'll address it in a later PR together with the code change

**Pull request merged**